ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/InMemory.hs

```diff
@@ -55,17 +57,59 @@ tests = testGroup "InMemory" [
         , testProperty "switchExpectedLedger"   prop_switchExpectedLedger
         , testProperty "pastAfterSwitch"        prop_pastAfterSwitch
         ]
-    , testGroup "Lookup" [
-          testProperty "ledgerDbPast"           prop_pastVsSpec
-        ]
     ]
 
 {-------------------------------------------------------------------------------
   Serialisation
 -------------------------------------------------------------------------------}
 
-prop_serialise_ChainSummary :: Trivial (ChainSummary Int TestBlock) -> Property
-prop_serialise_ChainSummary (Trivial summary) = prop_serialise summary
+-- | The LedgerDB is parametric in the ledger @l@. We use @Int@ for simplicity.
+example_ledger :: Int
+example_ledger = 100
+
+golden_ledger :: FlatTerm
+golden_ledger =
+    [ TkListLen 2
+      -- VersionNumber
+    , TkInt 1
+      -- ledger: Int
+    , TkInt 100
+    ]
+
+-- | The old format based on the @ChainSummary@. To remain backwards compatible
+-- we still accept this old format.
+golden_ChainSummary :: FlatTerm
+golden_ChainSummary =
+    [ TkListLen 3
+      -- tip: WithOrigin (RealPoint TestBlock)
+    , TkListLen 1
+    , TkListLen 2
+    , TkInt 3
+    , TkListBegin, TkInt 0, TkInt 0, TkBreak
+      -- chain length: Word64
+    , TkInt 10
+      -- ledger: Int for simplicity
+    , TkInt 100
+    ]
```

Comment:
I know it's just a unit test, but now I at least now for sure the decoder actually is backwards compatible.